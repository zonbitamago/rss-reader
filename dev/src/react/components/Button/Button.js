import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./Button.css";
import MaterialButton from "@material-ui/core/Button";

const StyleYes = {
  background: "#03A9F4",
  color: "white",
  borderRadius: "5px"
};

const StyleNo = {
  background: "#FF1744",
  color: "white",
  borderRadius: "5px"
};

class Button extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    if (this.props.type == "yes") {
      return (
        <div className="Button">
          <MaterialButton
            id="yes"
            onClick={this.props.handleClick}
            variant="raised"
            style={StyleYes}
          >
            Yes
          </MaterialButton>
        </div>
      );
    } else if (this.props.type == "no") {
      return (
        <div className="Button">
          <MaterialButton
            id="no"
            onClick={this.props.handleClick}
            variant="raised"
            style={StyleNo}
          >
            No
          </MaterialButton>
        </div>
      );
    }
  }
}

Button.propTypes = {
  type: PropTypes.string.isRequired,
  handleClick: PropTypes.func
};

Button.defaultProps = {};

export default Button;
