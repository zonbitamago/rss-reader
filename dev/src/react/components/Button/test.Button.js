import React from 'react'
import { shallow } from 'enzyme'

import Button from './Button'

describe('Button', () => {
  let component, props

  beforeEach(() => {
    props = {}
    component = shallow(<Button {...props} />)
  })

  it('should', () => {
    expect(component).toMatchSnapshot()
  })
})