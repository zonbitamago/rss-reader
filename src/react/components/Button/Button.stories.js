import React from "react";
import { storiesOf } from "@storybook/react";

import Button from "./Button";

storiesOf("Atoms/Button", module)
  .add("Yes", () => <Button type="yes" />)
  .add("No", () => <Button type="no" />);
