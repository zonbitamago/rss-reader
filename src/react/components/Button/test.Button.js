import React from "react";
import { shallow } from "enzyme";

import Button from "./Button";

describe("Button yes", () => {
  let component, props;

  beforeEach(() => {
    props = { type: "yes" };
    component = shallow(<Button {...props} />);
  });

  it("should", () => {
    expect(component).toMatchSnapshot();
  });
});

describe("Button no", () => {
  let component, props;

  beforeEach(() => {
    props = { type: "no" };
    component = shallow(<Button {...props} />);
  });

  it("should", () => {
    expect(component).toMatchSnapshot();
  });
});

describe("Button else", () => {
  let component, props;

  beforeEach(() => {
    props = { type: "else" };
    component = shallow(<Button {...props} />);
  });

  it("should", () => {
    expect(component).toMatchSnapshot();
  });
});
