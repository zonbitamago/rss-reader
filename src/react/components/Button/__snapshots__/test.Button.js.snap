// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button else should 1`] = `""`;

exports[`Button no should 1`] = `
<div
  className="Button"
>
  <WithStyles(Button)
    id="no"
    style={
      Object {
        "background": "#FF1744",
        "borderRadius": "5px",
        "color": "white",
      }
    }
    variant="raised"
  >
    No
  </WithStyles(Button)>
</div>
`;

exports[`Button yes should 1`] = `
<div
  className="Button"
>
  <WithStyles(Button)
    id="yes"
    style={
      Object {
        "background": "#03A9F4",
        "borderRadius": "5px",
        "color": "white",
      }
    }
    variant="raised"
  >
    Yes
  </WithStyles(Button)>
</div>
`;
